### PR TITLE
PT-1035 Fix management of warning objects

### PR DESCRIPTION
**Pull request by franck-fourel**

### CODE REVIEW

File: structure/base/base.go [hunk 1]
```diff
@@ -116,3 +120,20 @@ func (b *Base) WithReference(reference string) *Base {
 	}
 	return base
 }
+
+func (b *Base) WithReferenceNoWarning(reference string) *Base {
```
Comment (yvon-dblg) [inline structure/base/base.go]: This function seems to be unused. Was it really needed ?

File: structure/base/base.go [hunk 2]
```diff
@@ -116,3 +120,20 @@ func (b *Base) WithReference(reference string) *Base {
...
+	base := &Base{
```
Comment (yvon-dblg) [inline structure/base/base.go]: I know the rest of the code do the same, but:
```suggestion
	base := b.WithReference(reference)
```
without the duplicate code might be better ?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,10 @@
 # Platform 
 The Tidepool platform API
 
+## 0.6.2 2020-01-21
+### Fixed
+- PT-1035 Patch the PT-903 fix. Warning were not managed correctly when a warning was in the middle of the process. All the objects after the warnings were ignored.
```
Comment (yvon-dblg) [inline CHANGELOG.md]:
```suggestion
- PT-1035 Fix on the PT-903 patch. Warning were not managed correctly when a warning was in the middle of the process. All the objects after the warnings were ignored. 
```
Not sure on this one.

**Review by yvon-dblg — COMMENTED** [review]
Few questions

File: structure/base/base.go (hunk 1 above)
Comment (franck-fourel) [reply structure/base/base.go]: correct

File: structure/base/base.go (hunk 2 above)
Comment (franck-fourel) [reply structure/base/base.go]: this code is removed ;-)

**Review by yvon-dblg — APPROVED** [review]

**Pull request merged**